mmedit/models/losses/gan_loss.py

```diff
@@ -83,20 +88,189 @@ def forward(self, input, target_is_real, is_disc=False):
         Returns:
             Tensor: GAN loss value.
         """
+
         target_label = self.get_target_label(input, target_is_real)
         if self.gan_type == 'hinge':
             if is_disc:  # for discriminators in hinge-gan
                 input = -input if target_is_real else input
                 loss = self.loss(1 + input).mean()
             else:  # for generators in hinge-gan
                 loss = -input.mean()
+        elif self.gan_type == 'smgan':
+            mask = kwargs['mask']
+
+            input_height, input_width = input.shape[2:]
+            mask_height, mask_width = mask.shape[2:]
+
+            # Handle inconsistent size between outputs and masks
```

Comment:
@ckkelvinchan If I remember correctly, it is for handling the case when the `RandomResizeCrop` function in data augmentation yields a result with size which is off by 1 pixel (e.g. sometimes the shape of result is (511, 512) rather than exactly (512, 512))

Reply:
Why is that? Is it a bug or expected output?